tensorflow/compiler/xla/service/gpu/ir_emitter_unnested.cc

```diff
@@ -1956,26 +1956,26 @@ static void UnrollInnerTileLoop(
     bool check_x_tile_bounds, int64 x_num_steps, int64 step_x,
     int64 vector_size, const string& loop_name, KernelSupportLibrary* ksl,
     llvm::Value* start_offset_x, llvm::Value* y_loc, llvm::Value* tile_width,
-    IrArray::Index& source_idx, llvm::IRBuilder<>& b_,
+    const IrArray::Index& source_idx, llvm::IRBuilder<>* b_,
     const IrEmitterUnnested::EmitElementFunction* emit_elem_function) {
   llvm::Type* index_ty = tile_width->getType();
   auto constant = [&](int64 val) {
     return llvm::ConstantInt::get(index_ty, val);
   };
-  for (int j = 0; j < x_num_steps / vector_size; j++) {
-    for (int i = 0; i < vector_size; i++) {
+  for (int64 j = 0; j < x_num_steps / vector_size; j++) {
+    IrArray::Index source_idx_x_base =
```

Comment:
I think this can be moved outside of the loop. Note that AddOffsetToDim creates a new index and doesn't modify the existing one.